libs/wingsdk/src/util/util.ts

```diff
@@ -1,5 +1,39 @@
 import { Code, InflightClient } from "../core";
-import { Duration } from "../std";
+import { Duration, IResource } from "../std";
+
+/**
+ * Properties for `util.wait`.
+ */
+export interface WaitProps {
+  /**
+   * The timeout for keep trying predicate
+   * @default 1m
+   */
+  readonly timeout?: Duration;
+  /**
+   * Interval between predicate retries
+   * @default 0.1s
+   */
+  readonly interval?: Duration;
+}
+
+/**
+ * Represents a predicate with an inflight "handle" method that can be passed to
+ * `util.busyWait`.
+ * @inflight `@winglang/sdk.util.IPredicateHandlerClient`
+ */
+export interface IPredicateHandler extends IResource {}
+
+/**
+ * Inflight client for `IPredicateHandler`.
+ */
+export interface IPredicateHandlerClient {
```

Comment:
Curious - I am wondering if it is possible to avoid `IPredicateHandler` and just use `IPredicateHandlerClient` directly on the method...

Reply:
no

```
|       util.waitUntil((): bool => { 
   | /--------------------^
65 | |       invokeCounter.inc();
66 | |       throw("ERROR"); 
67 | |     });
   | \-----^ Expected type to be "IPredicateHandlerClient", but got "inflight (): bool" instead
```

Reply:
It would be useful to find a way to collapse these in the our `@winglang/sdk` source code (it is quite boilerplatey to have to define two interfaces)